src/_main_/utils/translation/__init__.py

```diff
@@ -106,46 +136,67 @@ def get_flattened_dict(self):
 
     def get_flattened_dict_for_excluded_keys(self):
         return self._excluded
-    
 
-    def translate(self, source_language: str, destination_language: str):
+
+    def translate (self, source_language: str, destination_language: str) -> Tuple[dict, List[str], List[dict]]:
         """
         Translate the flattened dictionary values from source_language to destination_language.
 
         Returns:
             dict: The translated dictionary in its original nested structure.
         """
-        # Flattened dictionary keys and values
-        keys = []
-        untranslated_text_entries = []
-        for key,value in self._flattened.items():
+
+        keys = [] # Flattened dictionary keys and values
+        untranslated_text_entries = [] # texts to be translated
+        text_hashes = [] # text hashes to be translated
+
+        for key, value in self._flattened.items():
```

Comment:
add a TODO: here to parallelize this.  maybe split the list into maybe 10 and have the job of the translation done in several threads and the results combined.